netebpfext/net_ebpf_ext_hook_provider.c

```diff
@@ -95,12 +110,17 @@ _ebpf_ext_attach_init_rundown(net_ebpf_extension_hook_client_t* hook_client)
  *
  */
 static void
-_ebpf_ext_attach_wait_for_rundown(_Inout_ net_ebpf_ext_hook_rundown_t* rundown)
+_ebpf_ext_wait_for_rundown(_Inout_ net_ebpf_ext_hook_rundown_t* rundown)
 {
     NET_EBPF_EXT_LOG_ENTRY();
 
-    ExWaitForRundownProtectionRelease(&rundown->protection);
-    rundown->rundown_occurred = TRUE;
+    if (rundown->rundown_initialized == TRUE) {
```

Comment:
It looks to me like this is the crux of the fix. I haven't been able to reason out why this resolves the issue though.

Based on the test failure crash dump hang, we are waiting (forever) on DriverUnload. In that code path, I can't find a place where we would be invoking this function on an uninitialized rundown object.